components/feature/accounts/src/main/java/mozilla/components/feature/accounts/FxaWebChannelFeature.kt

```diff
@@ -0,0 +1,332 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.accounts
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import mozilla.components.browser.session.SelectionAwareSessionObserver
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.browser.session.runWithSessionIdOrSelected
+import mozilla.components.concept.engine.Engine
+import mozilla.components.concept.engine.EngineSession
+import mozilla.components.concept.engine.webextension.MessageHandler
+import mozilla.components.concept.engine.webextension.Port
+import mozilla.components.concept.engine.webextension.WebExtension
+import mozilla.components.concept.sync.AuthType
+import mozilla.components.service.fxa.FxaAuthData
+import mozilla.components.service.fxa.SyncEngine
+import mozilla.components.service.fxa.manager.FxaAccountManager
+import mozilla.components.service.fxa.toAuthType
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import mozilla.components.support.base.log.logger.Logger
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import java.lang.ClassCastException
+import java.util.WeakHashMap
+
+/**
+ * Feature implementation that provides Firefox Accounts WebChannel support.
+ * For more information https://github.com/mozilla/fxa/blob/master/packages/fxa-content-server/docs/relier-communication-protocols/fx-webchannel.md
+ * This feature uses a web extension to communicate with FxA Web Content.
+ *
+ * Boilerplate around installing and communicating with the extension will be cleaned up as part https://github.com/mozilla-mobile/android-components/issues/4297.
+ *
+ * @property context a reference to the context.
+ * @property customTabSessionId optional custom tab session ID, if feature is being used with a custom tab.
+ * @property engine a reference to application's browser engine.
+ * @property sessionManager a reference to application's [SessionManager].
+ * @property accountManager a reference to application's [FxaAccountManager].
+ */
+@Suppress("TooManyFunctions")
+class FxaWebChannelFeature(
```

Comment:
the other class is called `Firefox Accounts Auth Feature` , this one just says `Fxa....`, does that matter?